File: framework-web/src/test/java/com/example/framework/web/it/error/TestController.java
package com.example.framework.web.it.error;

import com.example.framework.common.exceptions.ServiceException;
import com.example.framework.web.entity.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.ResponseBody;

/**
 * @author MaGuangZu
 * @since 2024-05-19
 */
@Controller
public class TestController {
	@GetMapping("/failed")
	public String failed() {
		throw new ServiceException("failed");
	}

	@ResponseBody
	@GetMapping("/failed2")
	public Result<Void> failed2() {
		throw new ServiceException("failed");
	}

	@ResponseBody
	@PostMapping("/failed3")
	public Result<Void> failed3() {
		return Result.ok();
	}

}

File: framework-web/src/test/kotlin/com/example/framework/web/it/error/TestController.kt
package com.example.framework.web.it.error

import com.example.framework.common.exceptions.ServiceException
import com.example.framework.web.entity.Result
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.ResponseBody

/**
 * @author MaGuangZu
 * @since 2023-09-08
 */
@Controller
class TestController {

	@GetMapping("/failed")
	fun failed(): String {
		throw ServiceException("failed")
	}

	@ResponseBody
	@GetMapping("/failed2")
	fun failed2(): Result<Void> {
		throw ServiceException("failed")
	}

}
